internal/controller/monovertexrollout/monovertexrollout_progressive.go

```diff
@@ -315,6 +315,9 @@ func scaleDownUpgradingMonoVertex(
 		// which is necessary when performing the health check for "ready replicas >= desired replicas"
 		// 2. that the total number of Pods (between the 2 monovertices) before and during upgrade remains the same
 		upgradingChildScaleTo := scaleValue.Initial - scaleValue.ScaleTo
+		if upgradingChildScaleTo <= 0 { // if for some reason the Initial value was 0, we don't want to set our Pods to 0
+			upgradingChildScaleTo = 1
+		}
```

Comment:
make sure we don't accidentally scale a new upgrading vertex to 0 Pods just because of the calculation - we need to have at least 1 Pod to test it